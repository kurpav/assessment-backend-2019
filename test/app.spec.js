const path = require('path');
const { createTestClient } = require('apollo-server-testing');
const { ApolloServer } = require('apollo-server-express');
const { importSchema } = require('graphql-import');
const { makeExecutableSchema } = require('graphql-tools');

const resolvers = require('../src/graphql/resolvers');

const GET_INCIDENT = `
query GetIncident($id: ID!) {
  incident(id: $id) {
    title
    assignee
  }
}
`;

describe('Incident management system', () => {
  const typeDefs = importSchema(
    path.join(__dirname, '../src/graphql/schemas/schema.graphql')
  );

  it('should return an incident', async () => {
    const _doc = {
      _id: 'id',
      title: 'Test title',
      description: 'Test description',
      assignee: 'test@gmail.com',
      status: 'Created'
    };
    resolvers.Query.incident = jest.fn().mockResolvedValue(_doc);
    const schema = makeExecutableSchema({ typeDefs, resolvers });
    const server = new ApolloServer({ schema });
    const { query } = createTestClient(server);

    const res = await query({ query: GET_INCIDENT, variables: { id: 'id' } });
    expect(res).toMatchSnapshot();
  });
});
