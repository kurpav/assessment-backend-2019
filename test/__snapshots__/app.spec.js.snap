// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incident management system should work 1`] = `
Object {
  "data": Object {
    "incident": Object {
      "assignee": "test@gmail.com",
      "title": "Test title",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
